packages/amplify-graphql-relational-transformer/src/has-many/has-many-directive-ddb-references-transformer.ts

```diff
@@ -73,5 +74,9 @@ export class HasManyDirectiveDDBReferencesTransformer implements DataSourceBased
     config.indexName = `gsi-${objectName}.${fieldName}`;
     config.referenceNodes = getReferencesNodes(config, context);
     validateReferencesBidirectionality(config);
+    const overrideIndexName = getOverrideIndexName(config);
+    if (overrideIndexName) {
+      config.indexName = overrideIndexName;
+    }
```

Comment:
1. Super nitpicky here & below: why separate this assignment from the config.indexName assignment 2 lines above, as opposed to a single block to handle the default & override cases?
2. Can you confirm that even though we're processing the GSI name in `hasOne` and `hasMany` transformers, it's actually created on the related `belongsTo` table?

Reply:
This asserts the GSI is created on the `belongsTo` table. https://github.com/aws-amplify/amplify-category-api/pull/2738/files#diff-acd0902518c587805b3f7217654188e3cc323f5dc9500f146db4f3673109b1f6R824